### PR TITLE
[14.0][MIG] account_invoice_transmit_method

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by alexis-via** [comment]
Depends on the migration PR of base_view_inheritance_extension https://github.com/OCA/server-tools/pull/1937

**Comment by simahawk** [comment]
@alexis-via 2 things:
1. please check the dependency error on OCA/server-tools#1937 so that we can speed it up
2. please squash !fixup commits (even if not yours :wink: )

File: account_invoice_transmit_method/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        "'&', ('supplier_ok', '=', transmit_method_domain_purchase),"
+        " ('supplier_ok', '=', True)]",
+    )
+    # Field used to match specific invoice transmit method
+    # to show/display fields/buttons, add constraints, etc...
+    transmit_method_code = fields.Char(
+        related="transmit_method_id.code", store=True, string="Transmission Method Code"
+    )
+    transmit_method_domain_sale = fields.Boolean(
+        compute="_compute_transmit_method_domain", default=True
```
Comment (jcoux) [inline account_invoice_transmit_method/models/account_move.py]: Useless default value.
```suggestion
        compute="_compute_transmit_method_domain"
```

File: account_invoice_transmit_method/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    )
+    transmit_method_domain_purchase = fields.Boolean(
+        compute="_compute_transmit_method_domain", default=True
```
Comment (jcoux) [inline account_invoice_transmit_method/models/account_move.py]: Useless default value.
```suggestion
        compute="_compute_transmit_method_domain"
```

File: account_invoice_transmit_method/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            else:
+                record.transmit_method_domain_sale = True
+                record.transmit_method_domain_purchase = True
+
+    @api.onchange("partner_id")
+    def _transmit_method_partner_change(self):
+        if self.partner_id and self.move_type:
+            if self.is_sale_document():
+                self.transmit_method_id = (
+                    self.partner_id.customer_invoice_transmit_method_id.id or False
```
Comment (jcoux) [inline account_invoice_transmit_method/models/account_move.py:59]: No need to set the `.id`, set the record set works too, and no need `or False`: it's already done by the ORM.
```suggestion
                    self.partner_id.customer_invoice_transmit_method_id
```

File: account_invoice_transmit_method/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                )
+            elif self.is_purchase_document():
+                self.transmit_method_id = (
+                    self.partner_id.supplier_invoice_transmit_method_id.id or False
```
Comment (jcoux) [inline account_invoice_transmit_method/models/account_move.py:63]: No need to set the `.id`, set the record set works too, and no need `or False`: it's already done by the ORM.
```suggestion
                    self.partner_id.supplier_invoice_transmit_method_id
```

**Review by jcoux — CHANGES_REQUESTED** [review]

**Comment by patrickrwilson** [comment]
@alexis-via Thanks for your work on this, I've tested and it works well. 

Can you squash all the automated commits (The Translation, [UPD] and .pot) to cleanup the commit history. Once that is done I'll review. Thanks.

**Review by TDu — APPROVED** [review]
LG, thank you for migrating this and its dependencies.
I only see non blocking comments on my side.

**Comment by qpeyramale** [comment]
LGTM

**Comment by patrickrwilson** [comment]
@alexis-via just need automated commit history squashed then I will approve.

**Review by jcoux — APPROVED** [review]
Thanks for changes.
My suggestion about `.id or False` have not been addressed, but it's not blocking!
Approved :+1:

**Pull request merged**